### PR TITLE
Port e2e/scheduler_predicates.go to e2e Framework.

### PR DESCRIPTION
**Pull request by ghost**
Partially addresses #13503, #13574, #13828 

cc: @derekwaynecarr @fgrzadkowski @gmarek

### CODE REVIEW

**Comment by quinton-hoole** [comment]
Added e2e-not-required label, as these scheduler_predicates tests cannot run in parallel with other tests, so are not run as part of the automated Jenkins GDE e2e run, or the one that merge-bot runs.

**Comment by quinton-hoole** [comment]
Also partially addresses #13639.

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -136,30 +136,17 @@ func cleanupPods(c *client.Client, ns string) {
 }
 
 var _ = Describe("SchedulerPredicates", func() {
+	framework := Framework{BaseName: "sched-pred"}
```
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: You probably want to extend timeout for namespace deletion. This test saturates the cluster by creating a lot of pods, and thus events. I added it recently: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework.go#L41

**Comment by gmarek** [comment]
Only one thing - except that LGTM.

File: test/e2e/scheduler_predicates.go (hunk 1 above)
Comment (quinton-hoole) [inline test/e2e/scheduler_predicates.go]: The current version of the test (before this PR) does not have a timeout on namespace deletion.  I'd prefer to add the timeout in a separate PR, if it's necessary.  Besides that I don't currently know what a good value for that timeout would be, and don't want to shove an arbitrary one in for no good reason.  OK @gmarek ?

Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: OK. We generally put 1h for 'long timeout' (in density and load tests).

**Comment by quinton-hoole** [comment]
Shippable was a flake. I kicked her.

**Pull request merged**